src/ScenarioPreview/ScenarioPreview.test.tsx

```diff
@@ -0,0 +1,151 @@
+    render(
+      <ScenarioPreview
+        name={Default.args.name}
+        href={Default.args.href}
+        image={Default.args.image}
+        description={Default.args.description}
+        format={Default.args.format}
+        formatVersion={Default.args.formatVersion}
+        file={Default.args.file}
+        roadLabel="Test Road Name"
+      />
+    );
+
+    const roadLabelElement = screen.getByTestId("road-label");
+    expect(roadLabelElement).toBeInTheDocument();
+    expect(roadLabelElement).toHaveTextContent("Test Road Name");
+  });
+
+  // Test that the divider renders when optional props are present
+  test("renders divider when optional props are provided", () => {
+    const { container } = render(
+      <ScenarioPreview
+        name={Default.args.name}
+        href={Default.args.href}
+        image={Default.args.image}
+        description={Default.args.description}
+        format={Default.args.format}
+        formatVersion={Default.args.formatVersion}
+        file={Default.args.file}
+        createdAt={Default.args.createdAt}
+        user={Default.args.user}
+        roadLabel={Default.args.roadLabel}
+      />
+    );
+
+    expect(container.querySelector(".MuiDivider-root")).toBeInTheDocument();
+  });
+
+  // Test that no divider renders when optional props are not provided
+  test("does not render divider when optional props are missing", () => {
+    const { container } = render(
+      <ScenarioPreview
+        name={WithoutOptionalProps.args.name}
+        href={WithoutOptionalProps.args.href}
+        image={WithoutOptionalProps.args.image}
+        description={WithoutOptionalProps.args.description}
+        format={WithoutOptionalProps.args.format}
+        formatVersion={WithoutOptionalProps.args.formatVersion}
+        file={WithoutOptionalProps.args.file}
+      />
+    );
+
+    expect(container.querySelector(".MuiDivider-root")).not.toBeInTheDocument();
+  });
+
+  // Test for handling overflow text in name, description, and road label
+  test("handles overflow text correctly", () => {
+    render(
+      <ScenarioPreview
+        name={WithOverflowText.args.name}
+        href={WithOverflowText.args.href}
+        image={WithOverflowText.args.image}
+        description={WithOverflowText.args.description}
+        format={WithOverflowText.args.format}
+        formatVersion={WithOverflowText.args.formatVersion}
+        file={WithOverflowText.args.file}
+        roadLabel={WithOverflowText.args.roadLabel}
+      />
+    );
+
+    expect(screen.getByText(WithOverflowText.args.name)).toBeInTheDocument();
+    expect(
+      screen.getByText(WithOverflowText.args.description)
+    ).toBeInTheDocument();
+    expect(screen.getByText(WithOverflowText.args.file)).toBeInTheDocument();
+    expect(
+      screen.getByText(WithOverflowText.args.roadLabel)
+    ).toBeInTheDocument();
+  });
+
+  // Test that all properties are rendered correctly
+  test("renders all provided properties correctly", () => {
```

Comment:
Probably would be a good idea to also check that the image component exists/renders as part of this test